### ISSUE TITLE
Support more merge modes

### ISSUE THREAD

**Issue by maxpumperla** [deeplearning4j/deeplearning4j#3915]
#### Issue Description

In model import, mapping merge modes from keras, there are a few of them missing: 

https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/layers/KerasMerge.java#L92-L98

After syncing with @AlexDBlack, `sub`, `ave` and `max` should be easy to support on our side. (`cos` is gone from keras 2, `dot` is less of a low-hanging fruit).

Once this is integrated, I'll create a follow-up ticket for model import.
---

Support:
- [ ] `ave`
- [ ] `max`
- [ ] `sub`

**Comment by eraly** [comment]
We already have element wise vertex operations that do add. So adding ave is trivial.
I don't think we have a very straightforward to do element wise max. I can think of workarounds that are inefficient. Will sync with Alex.

**Comment by AlexDBlack** [comment]
Are we talking pair-wise max only? Or general N-input-arrays max? The former case is a bit easier.

The latter case: my first thought is concat + max along dimension for forward pass.
Then for backward pass, concat as per forward, apply IsMax op on concatted array, broadcast muli of the epsilons on the concatted array, get op to get the epsilons out.

**Comment by eraly** [comment]
I actually don't know if the keras keras.layers.merge.Maximum() is a reduction operation like the one we get with the max along the minibatch dimension. 

eg. merge the output of two dense layers with nin=2 and minibatch size of 2
say [[1,12],[3,14]] and [[11,2],[13,4]] 
should give [[11,12],[13,14]] or [13,14]?

@maxpumperla?

**Comment by eraly** [comment]
I didn't answer to pairwise or general N-input. It's definitely any number of inputs.

**Comment by maxpumperla** [comment]
@eraly @AlexDBlack it's n-dim element-wise max and average for arbitrary tensors with matching shape, e.g. for four layers to be merged it would be

Four example inputs to merge layer:
```
[1,0,0,0],
[0,2,0,0],
[0,0,3,0],
[0,0,0,4]
```

max:
```
[1,2,3,4]
```

avg:
```
[0.25,0.5,0.75,1.0]
```

**Comment by eraly** [comment]
Thanks @maxpumperla

**Comment by maxpumperla** [comment]
@AlexDBlack @eraly quick update: as of version 2.0.8 keras also has `subtract` as a new mode.

**Comment by eraly** [comment]
We have element wise subtraction supported already

Edit: In Keras the link I found is with pairs which is what we currently have. No support for 'n' number of elements in the list. Don't think it's needed currently.

https://github.com/fchollet/keras/pull/7573/files/43328d9d62990672e9aecb0507d3aa90aa22d0b1#diff-bc3a72e8510dc2a86e31fad80dff6c3cR214

**Comment by eraly** [comment]
PR open for max and average. We already have sub.

**Comment by maxpumperla** [comment]
cool! thanks for that

**Comment by AlexDBlack** [comment]
https://github.com/deeplearning4j/deeplearning4j/pull/4036
